Release 1.13.0 (#1250)
heremaps/here-data-sdk-cpp: CHANGELOG.md

## v1.13.0 (27/09/2021)

**Common**

* Added a new `olp::client::EqualJitterBackdownStrategy` backdown strategy. You can use it to define the waiting time between retries in `olp::client::RetrySettings`.
* Changed the open behavior for the protected cache. Now, it tries to open the cache in the r/w mode if it has enough filesystem permissions, and the user did not explicitly instruct to open the cache in the r/o mode via `olp::cache::CacheSettings::openOptions`.

> * Changed the `olp::cache::DefaultCache::Open` method. Now, it returns `olp::cache::DefaultCache::StorageOpenResult::ProtectedCacheCorrupted`  if you try to open a broken database, specifically more than 4 L0 levelDB storage files, and you need to compact the database manually before you try to open it again.

Reviewer: ```suggestion
* Changed the `olp::cache::DefaultCache::Open` method. Now, it returns `olp::cache::DefaultCache::StorageOpenResult::ProtectedCacheCorrupted` if you try to open a broken database, specifically more than 4 L0 levelDB storage files, and you need to compact the database manually before you try to open it again.
```

* Added a bool operator to ApiResponse class, a shortcut to `ApiResponse::IsSuccessful()`.
* Removed arbitrary 2s delay on shutdown.
* Now request isn't sent if bearer token is empty. Previously, if authentication failed for some reason, data request was still sent with empty token.
* Fixed WinHttp request was not freed correctly after an error.
* Fixed unicode compilation on windows.

> * Fixed `olp::Dir::FileExist()` method for windows.

Reviewer 2: ```suggestion
* Fixed `olp::Dir::FileExist()` method when compiled for Windows. Instead of checking the `FILE_ATTRIBUTE_NORMAL` attribute it now checks that the attributes are unequal to `INVALID_FILE_ATTRIBUTES` to assume a file is present on the filesystem.
```

**olp-cpp-sdk-authentication**

> * Improved use of server time. Now we always retry with a server time on a wrong timestamp error. Also server time is incremented on retry.

Reviewer 2: ```suggestion
* Improved usage of local time and server time. When the local time is off and backend returns a wrong timestamp error, we try again by using the server time parsed from the response header field `Date`. Additionally, the server time is incremented locally between retries by adding the elapsed time elapsed between request and response.
```

> * Fixed the various APIs in authentication component to use the server time when the `use_system_time` option is false.

Reviewer 2: ```suggestion
* Fixed the various authentication component APIs to use the server time if the `use_system_time` option is set to `false`.
```

**olp-cpp-sdk-dataservice-read**

> * Fixed request repetition if the same request in other thread have ended up with an error. In that case the error will be shared.

Reviewer 2: ```suggestion
* Fixed unnecessary request repetition if the same blob request ended up with an error while multiple threads are waiting for it. Now the returned error will be shared among waiting threads, and does not result as before that each waiting thread request the same blob again and again, unaware of the error.
```

> ##  v1.12.0 (03/06/2021)

Reviewer: ```suggestion
## v1.12.0 (03/06/2021)
```
